### ISSUE TITLE
Validation taint-escape doesn't apply to `Request::array()` and `Request::collect()` accessors

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#840]
## Summary

The `KEYED_ACCESSOR_METHODS` list in `ValidationTaintHandler` covers only `validated`, `input`, `string`, `str`. Laravel 11+ added two sibling accessors that read from the same data pool:

- `Request::array($key)` — returns `array<int|string, mixed>`
- `Request::collect($key)` — returns `\Illuminate\Support\Collection`

Both are stubbed with `@psalm-taint-source input` (`stubs/common/Support/Traits/InteractsWithData.stubphp` lines 133, 143). Because they are not in `KEYED_ACCESSOR_METHODS`, the per-rule taint-escape never applies to their return values, so reads after `$request->validate(['email.*' => 'email'])` (or a typed `FormRequest` with the same rule) keep full taint and produce false-positive `TaintedHeader` / `TaintedCookie` reports at downstream sinks like `MailMessage::cc()` / `redirect()->to()`.

## Repro

```php
use Illuminate\Http\Request;

/** @psalm-suppress MixedArgument */
function massInvite(Request $request): void
{
    $request->validate(['email.*' => 'email']);

    foreach ($request->array('email') as $email) {
        // TaintedHeader: rule's escape never applied because `array` is not in KEYED_ACCESSOR_METHODS
        Mail::to($recipient)->cc($email)->send($notification);
    }
}
```

## Real-world hit

Surfaced on a large Laravel project (IxDF, ~14k PHP files) with a controller that follows the idiomatic Laravel 11+ pattern:

```php
$emails = $request->array('email');
$results = $this->addMembersAsTeamManager->execute($team, $emails, ...);
```

Two `TaintedHeader` reports flow from `Request::array` through several function calls to `MailMessage::cc()`. Both are false positives — the validation rule constrains each element to be an email (header-safe).

## Why #838 (PR #839) didn't address this

\#838 fixed wildcard-suffix handling for indexed access via `input()`. The `array()` / `collect()` accessor is a separate code path entirely:

- `ValidationTaintHandler::matchKeyedAccessor` rejects the call early because the method name isn't in `KEYED_ACCESSOR_METHODS`.
- The rule lookup (helped by the new `lookupRuleByKey`) never runs.
- No taint escape is applied.

## Proposed fix

Add `'array'` and `'collect'` to `KEYED_ACCESSOR_METHODS`. Existing logic should otherwise apply unchanged:

- `matchKeyedAccessor` already requires a single literal-string-key first arg with no second argument — `array($key)` / `collect($key)` match this shape.
- `ValidationRuleAnalyzer::lookupRuleByKey` (#839) already handles whole-array (`'email.*'` → `'email'`) and indexed (`.0`) lookups.
- `InlineValidateRulesCollector::resolveEscapeFromAccessorRhs` shares the same const, so `$arr = $request->array('email')` variable bindings should pick up the escape automatically.

One subtlety to verify: when `$request->array('email')` returns `array<int, string>` and a `foreach` extracts elements, Psalm's `arrayvalue-fetch` step propagates taint per element. Since the escape is applied at the call return, the taint that flows into elements should already be reduced.

### Tests to add

Mirror the existing `SafeInlineValidateStringAccessorEscapesHeader.phpt` shape:

- `SafeInlineValidateArrayAccessorEscapesHeader.phpt` — `validate(['email.*' => 'email'])` + `foreach ($request->array('email') as $e) cc($e);` — expect only `TaintedSSRF`.
- `SafeInlineValidateCollectAccessorEscapesHeader.phpt` — same with `$request->collect('email')->each(...)`.
- `SafeFormRequestArrayAccessorWildcardEscapesHeader.phpt` — typed FormRequest with `'email.*' => Rule::email()` + `$req->array('email')`.
- A scalar-rule variant: `validate(['email' => 'email'])` + `$request->array('email')`. At runtime Laravel wraps the scalar in a single-element array, so the same rule's escape should apply.

### Out of scope (separate consideration)

- `Request::input()` / `Request::array()` with explicit cast methods (`integer`, `boolean`, `date`, `enum`) — these are not taint sources, no rule lookup needed.
- `Request::only(...)` / `Request::except(...)` — these select multiple keys; matching them against a single rule needs different handling.

## Related

- #831 initial inline `$request->validate([...])` taint-escape support
- #834 / #837 variable-binding extension
- #838 / #839 wildcard-array indexed-access fallback

**Comment by alies-dev** [comment]
Resolved by #841 (merged). PR body declared `Closes #840` but GitHub auto-close did not fire.
